security-admin/src/main/java/org/apache/ranger/rest/PublicAPIs.java

```diff
@@ -415,4 +416,38 @@ public RangerManagementConfig getManagementConfiguration(@Context HttpServletReq
 		return rangerManagementConfig;
 	}
 
+	@GET
+	@Path("/admin/ok")
+	public String checkRangerHealth (@Context HttpServletRequest request) {
+
+		if (logger.isDebugEnabled()) {
+			logger.debug("==> PublicAPIs.checkRangerHealth(): ");
+		}
+		HashMap<String, String> responseMap = new HashMap<>();
+
+		// Default Response Parameters
+		String serviceStatus = "ok";
+		String dbStatus = "ok";
+		String version = RangerVersionInfo.getVersion();
+
+		// Get Database Version
+		String dbVersion = daoMgr.getXXDataHist().getDBVersion();
+
+		if (dbVersion.equalsIgnoreCase("Not Available")) {
+			serviceStatus = "partially running";
```

Comment:
Can you make default to be non-running, and set the version in json. I guess getDBVersion should throw an exception if it fails? String matching is error-prone.

Reply:
getDBVersion throws an exception in the function if it fails but the exception is handled inside the function. The function returns either "Not Available" or a version number.